### DISCUSSION TITLE
Support unicode string delimiters such as « »

### DISCUSSION THREAD

**Discussion by bblfish** [scala/scala3#17666] [Feature Requests]
The string delimiters such as """ ... """ get very confusing as soon as one has to use " in the text. For example for [Signing HTTP messages](https://www.ietf.org/archive/id/draft-ietf-httpbis-message-signatures-15.html#create-sig-input) one often has to produce strings like 

```
"example-dictionary";key="foo"
```

In Scala this must be written as a sequence starting with 4 quotes.

```scala
s""""example-dictionary";key="$arg""""
```

I think everyone will agree that that is not very readable for scala
developers, and even less when showing code snippets on github
repository when filing bugs against specs.

Much more readable would be
```scala
s«"example-dictionary";key="$arg"»
```

The idea would not be to remove the triple quotes which have the advantage of 
not requiring unicode input, but of adding the unicode ones for situations where
readability is important.

  There are a number of other unicode quotation symbols that could be used, and I 
don't want to make a decision on which of them is best, or if they should all be used, or if they should be used with different properties.  But something would be better than """

**Comment by SethTisue** [comment]
is there precedent for this in other languages?

**Comment by TheElectronWill** [comment]
UML uses guillemets for [stereotypes](https://www.uml-diagrams.org/stereotype.html).
I don't have another example :sweat_smile:

**Comment by som-snytt** [comment]
There is a classic thread on this topic that may date to the old mailing lists. I remember having such hopes for guillemets.
